test/components/connect.spec.js

```diff
@@ -949,6 +949,38 @@ describe('React', () => {
       expect(decorated.refs.wrappedInstance.someInstanceMethod()).toBe(someData)
     })
 
+    it('should do no parsing when the response body is empty', (done) => {
+      window.fetch = (request) => {
+        return new Promise((resolve) => {
+          resolve(new window.Response('', { status: 204 }))
+        })
+      }
+
+      @connect(() => ({ testFetch: `/empty` }))
+      class Container extends Component {
+        render() {
+          return <Passthrough {...this.props} />
+        }
+      }
+
+      const container = TestUtils.renderIntoDocument(
+        <Container />
+      )
+
+      const init = TestUtils.findRenderedComponentWithType(container, Container)
+      expect(init.state.data.testFetch).toIncludeKeyValues(
+        { fulfilled: false, pending: true, reason: null, refreshing: false, rejected: false, settled: false, value: null }
+      )
+
+      setImmediate(() => {
+        const fulfilled = TestUtils.findRenderedComponentWithType(container, Container)
+        expect(fulfilled.state.data.testFetch).toIncludeKeyValues(
+          { fulfilled: true, pending: false, reason: null, refreshing: false, rejected: false, settled: true, value: null }
+        )
+        done()
+      })
+    })
+
```

Comment:
The failed assertion is

> Uncaught Error: expected { BIG OBJECT } to include { 'fulfilled': true }

... but the actual error is buried inside BIG OBJECT:

> ... reason: [SyntaxError: Unexpected end of input] ...

--

> Uncaught Error: expected { fulfilled: false, meta: { request: { _bodyInit: undefined, _bodyText: '', credentials: 'same-origin', headers: { map: { accept: [Object], 'content-type': [Object] } }, method: 'GET', mode: null, referrer: null, url: '/empty' }, response: { _bodyInit: '', _bodyText: '', bodyUsed: true, headers: { map: { 'content-length': [Object], 'content-type': [Object] } }, ok: true, status: 200, statusText: undefined, type: 'default', url: '' } }, pending: false, reason: [SyntaxError: Unexpected end of input], refreshing: false, rejected: true, settled: true, value: null } to include { 'fulfilled': true }